spring-test/src/main/java/org/springframework/test/context/TestContext.java

```diff
@@ -109,6 +109,26 @@ default void publishEvent(Function<TestContext, ? extends ApplicationEvent> even
 	 */
 	Object getTestInstance();
 
+	/**
+	 * Tests whether a test method is part of this test context. Returns
+	 * {@code true} if this context has a current test method, {@code false}
+	 * otherwise.
+	 *
+	 * <p>The default implementation of this method always returns {@code false}.
+	 * Custom {@code TestContext} implementations are therefore highly encouraged
+	 * to override this method with a more meaningful implementation. Note that
+	 * the standard {@code TestContext} implementation in Spring overrides this
+	 * method appropriately.
+	 *
+	 * @return {@code true} if the test execution has already entered a test
+	 * method
+	 * @since 5.3
+	 * @see #getTestMethod()
+	 */
+	default boolean hasTestMethod() {
```

Comment:
Can you expand why this is needed in the context of this change?

Reply:
Because of https://github.com/spring-projects/spring-framework/blob/735b29c101438be8a8f24ca9ba0ec426cb9f4c3b/spring-test/src/main/java/org/springframework/test/context/transaction/TestContextTransactionUtils.java#L256-L261

and 

https://github.com/spring-projects/spring-framework/blob/735b29c101438be8a8f24ca9ba0ec426cb9f4c3b/spring-test/src/main/java/org/springframework/test/context/support/DefaultTestContext.java#L170-L174

Previously, the test context was always a method. Therefore, calling `testContext.getTestMethod()` was no problem, because there is always a test method. But now processing can already happen before the test reaches a test method. In that case, the assertion `Assert.state(testMethod != null, "No test method");` fails.  Because I could not predict the consequences of removing the assertion, I added that method.